contrib/bindings/go/pathrs/pathrs.go

```diff
@@ -0,0 +1,239 @@
+// Package pathrs is safe path resolution on Linux.
+// A binding for rust library with a consequent name.
+// It provides primitives to handle path resolving in a great manner.
+//
+// The configure, intofd, fromfd and duplicate api is not realized yet
+//
+// Copyright (C) 2019, 2020 Aleksa Sarai <cyphar@cyphar.com>
+// Copyright (C) 2019, 2020 SUSE LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pathrs
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+	"unsafe"
+)
+
+// Open opens directory as a root directory
```

Comment:
All of these doc comments are ... not really that great. But I can rewrite them after merging so you don't need to worry about that for now.

Reply:
Definitely, let's say they are not in a good shape
:disappointed_relieved:

Reply:
Don't sweat it, I can clean it up myself.